rubicon/objc/runtime.py

```diff
@@ -619,61 +619,69 @@ def should_use_fpret(restype):
         return False
 
 
-def send_message(receiver, selector, *args, restype=c_void_p, argtypes=None):
+def send_message(receiver, selector, *args, restype, argtypes, varargs=None):
     """Call a method on the receiver with the given selector and arguments.
 
     This is the equivalent of an Objective-C method call like ``[receiver sel:args]``.
 
-    :param receiver: The object on which to call the method. This may be an Objective-C object
-        (as an :class:`ObjCInstance`, :class:`objc_id`, or :class:`~ctypes.c_void_p`),
-        or an Objective-C class name (as a :class:`str` or :class:`bytes`).
+    .. note::
+
+        Some Objective-C methods take variadic arguments (varargs), for example `+[NSString stringWithFormat:]
+        <https://developer.apple.com/documentation/foundation/nsstring/1497275-stringwithformat?language=objc>`_.
+        When using :func:`send_message`, variadic arguments are not passed as regular arguments in the ``args`` list,
+        but as a separate ``varargs`` parameter.
+
+        This protects against accidentally passing too many arguments into a method and having the extra arguments
+        treated as varargs, even though the method doesn't actually take varargs.
```

Comment:
This doesn't describe how types are handled for the variadic arguments. Reading between the lines, I'm guessing it's a bit of "throw it over the fence and hope it works" - but that's a function of Objective C, not something we're in any position to fix.

Reply:
Right, it would be good to explain that in more detail. `varargs` are converted by `ctypes` using its default type mappings (i. e. Python `int`s become C `int`s, Python `bytes` become C `char *`s, etc.), but I would recommend using explicit casts/constructors if at all possible, so that `ctypes` doesn't accidentally do a different conversion than you expected.